core/shared/src/main/scala/org/http4s/headers/Deprecation.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package headers
+
+import cats.parse.Parser
+import org.typelevel.ci._
+
+object Deprecation {
```

Comment:
For the deprecation header I could find what seems to be an RFC _draft_ https://datatracker.ietf.org/doc/html/draft-ietf-httpapi-deprecation-header-02, was this approved or is it common enough? If so, could we add a link to the specification being adhered to?

Reply:
A current version is posted at [greenbytes](https://greenbytes.de/tech/webdav/draft-ietf-httpapi-deprecation-header-latest.html), which is curiously not synchronized with the IETF draft, which has expired.  As of August 2022, they are shifting their attention to a [Lifecycle field](https://mailarchive.ietf.org/arch/browse/httpapi/?q=draft-ietf-httpapi-deprecation-header).

We usually prefer standard headers, but this one is:
1. Useful
2. Has no standardized alternative
3. Is at least as standardized as some ad hoc things we find on MDN
4. Is unlikely to change.

I think we should proceed, but I do agree with linking to the standard, such as it is.